pkg/agent/eventsync.go

```diff
@@ -22,16 +23,25 @@ type EventSync struct {
 
 // NewEventSync creates an EventSync
 func NewEventSync() (*EventSync, error) {
-	migrations := []migrate.Migrator{
-		migrate.NewAttributeMigration(agent.apicClient, agent.cfg),
-	}
+	migrations := []migrate.Migrator{}
+
+	// Make sure only DA and Governance agents run migration processes
+	runMigrations := agent.cfg.GetAgentType() != config.TraceabilityAgent
 
+	// Check if marketplace is enabled
 	isMpEnabled := agent.agentFeaturesCfg != nil && agent.agentFeaturesCfg.MarketplaceProvisioningEnabled()
 
-	if isMpEnabled {
-		marketplaceMigration := migrate.NewMarketplaceMigration(agent.apicClient, agent.cfg, agent.cacheManager)
-		agent.marketplaceMigration = marketplaceMigration
-		migrations = append(migrations, marketplaceMigration)
+	if runMigrations {
+		// add attribute migration to migrations
+		attributeMigration := migrate.NewAttributeMigration(agent.apicClient, agent.cfg)
+		migrations = append(migrations, attributeMigration)
+
+		if isMpEnabled {
+			// add marketplace migration to migrations
+			marketplaceMigration := migrate.NewMarketplaceMigration(agent.apicClient, agent.cfg, agent.cacheManager)
+			agent.marketplaceMigration = marketplaceMigration
+			migrations = append(migrations, marketplaceMigration)
+		}
 	}
 
 	mig := migrate.NewMigrateAll(migrations...)
```

Comment:
If you want, you can also not pass `mig` to the discovery cache too. But since this is creating a migration with an empty slice, it will be the same behavior. It is up to you